python/lsst/meas/algorithms/subtractBackground.py

```diff
@@ -0,0 +1,260 @@
+    )
+    approxOrderX = pexConfig.Field(
+        doc="Approximation order in X for background Chebyshev (valid only with useApprox=True)",
+        dtype=int, default=6,
+    )
+    # Note: Currently X- and Y-orders must be equal due to a limitation in math::Chebyshev1Function2
+    # The following is being added so that the weighting attribute can also be configurable for the
+    # call to afwMath.ApproximateControl
+    approxOrderY = pexConfig.Field(
+        doc="Approximation order in Y for background Chebyshev (valid only with useApprox=True)",
+        dtype=int, default=-1,
+    )
+    weighting = pexConfig.Field(
+        doc="Use inverse variance weighting in calculation (valid only with useApprox=True)",
+        dtype=bool, default=True,
+    )
+
+    def validate(self):
+        pexConfig.Config.validate(self)
+        # Allow None to be used as an equivalent for "NONE", even though C++ expects the latter.
+        if self.algorithm is None:
+            self.algorithm = "NONE"
+
+class SubtractBackgroundTask(pipeBase.Task):
+    ConfigClass = SubtractBackgroundConfig
+    _DefaultName = "subtractBackground"
+
+    def run(self, exposure, background=None, stats=True, statsKeys=None):
+        """!Fit and subtract the background of an exposure
+
+        @param[in,out] exposure  exposure whose background is to be subtracted
+        @param[in,out] background  initial background model already subtracted from exposure
+            (an lsst.afw.math.BackgroundList). May be None if no background has been subtracted.
+        @param[in] stats  if True then measure the mean and variance of the full background model
+                        and record the results in the exposure's metadata
+        @param[in] statsKeys  key names used to store the mean and variance of the background
+            in the exposure's metadata (a pair of strings); if None then use ("BGMEAN", "BGVAR");
+            ignored if stats is false
+
+        @return an lsst.pipe.base.Struct containing:
+        - background  full background model (initial model with changes), an lsst.afw.math.BackgroundList
+
+        @throw RuntimeError if fitBackground returns a background of None
+        """
+        displayBackground = lsstDebug.Info(__name__).displayBackground
+
+        if background is None:
+            background = afwMath.BackgroundList()
+
+        maskedImage = exposure.getMaskedImage()
+
+        fitBg = self.fitBackground(maskedImage)
+        if not fitBg:
+            raise RuntimeError("Unable to estimate background for exposure")
+
+        fitBgImg = fitBg.getImageF()
+        maskedImage -= fitBgImg
+        background.append(fitBg)
+        del fitBgImg
+
+        if displayBackground > 1 or stats:
+            netBgImg = background.getImage()
+
+        if displayBackground > 1:
+            ds9.mtv(netBgImg, title="background", frame=3)
+
+        # Record statistics of the background in the bgsub exposure metadata.
```

Comment:
To say more about this: it seems very likely to me that most of our exposures will end up with metadata that contains statistics for the initially estimated background, but those values will not match the final background that was subtracted. Ick.